### PR TITLE
TECH-7750 tweak to unprompt

### PR DESCRIPTION
**Pull request by dataxerik**
[TECH-7750 <name here>](https://industrydive.atlassian.net/browse/TECH-7750)

### Technical Context
I mainly tested this for the `Add Source` page.

### Manual Testing Instructions
- Things that don't prompt 
  - [x] Reloading the page when it's not touched doesn't create a prompt
  - [x] Adding a blank interaction
- Things that do prompt
  - [x] Typing something into a input field 
  - [ ] Typing something into a textarea
  - [x] Selecting something from a dropdown
  - [x] Adding/Removing something from a slector
  - [x] Using a date or time picker

#### Developer

- [x] Ensure code complies with the [style guide](https://industrydive.atlassian.net/wiki/spaces/TECH/pages/639500407/Tech+Team+s+Style+Guides)
- [x] Devise edge cases to test the bug fix or feature being merged
- [x] Make sure the code is clean and understandable
- [x] Ensure code is not overly inefficient
- [x] Ensure documentation is understandable and accurate, including:
    - [x] Code comments and doc strings
    - [ ] ~Technical documentation~

#### Reviewer

- [ ] Ensure code complies with the [style guide](https://industrydive.atlassian.net/wiki/spaces/TECH/pages/639500407/Tech+Team+s+Style+Guides)
    - If there are few or minor issues, add inline GitHub comments
    - If there are many issues or it seems like the developer did not follow the style guide, leave a comment asking them to do so
- [ ] Devise edge cases to test the bug fix or feature being merged
- [ ] Make sure the code is clean and understandable
    - [ ] Ask for inline comments on unclear sections
- [ ] Ensure code is not overly inefficient
- [ ] Ensure documentation is understandable and accurate, including:
    - [ ] Code comments and doc strings
    - [ ] Technical documentation

### CODE REVIEW

File: media/js/admin/sources/promptUnsavedChanges.js [hunk 1]
```diff
@@ -20,30 +20,35 @@
...
-  customInputSetter();
-  setFormModifiedEvent();
-
   // The selector-chosen fields are inline and aren't available even if the document is ready, so need to wait until the load event
   $(window).on("load", () => {
-    $('.selector-chosen').bind("DOMSubtreeModified", () => {
-      formModified = true;
+    setFormModifiedEvent();
+
+    $('.selector-chosen').bind("DOMSubtreeModified", function () {
```
Comment (greglinch) [inline media/js/admin/sources/promptUnsavedChanges.js]:
```suggestion
    $('.selector-chosen').bind('DOMSubtreeModified', function () {
```

File: media/js/admin/sources/promptUnsavedChanges.js [hunk 2]
```diff
@@ -20,30 +20,35 @@
...
       formModified = true;
     });
   }
 
-  // Initial Setup
-  customInputSetter();
-  setFormModifiedEvent();
-
   // The selector-chosen fields are inline and aren't available even if the document is ready, so need to wait until the load event
   $(window).on("load", () => {
```
Comment (greglinch) [inline media/js/admin/sources/promptUnsavedChanges.js]:
```suggestion
  $(window).on('load', () => {
```

File: media/js/admin/sources/promptUnsavedChanges.js [hunk 3]
```diff
@@ -20,30 +20,35 @@
     Object.defineProperty(HTMLInputElement.prototype, "value", descriptor);
```
Comment (greglinch) [inline media/js/admin/sources/promptUnsavedChanges.js]:
```suggestion
    Object.defineProperty(HTMLInputElement.prototype, 'value', descriptor);
```

**Review by greglinch — CHANGES_REQUESTED** [review]
@dataxerik Nice improvements, thank you! Unfortunately, the dropdowns and text area on `Interaction` inline still don't prompt.

**Comment by greglinch** [comment]
Follow up ticket here for additional fixes https://industrydive.atlassian.net/browse/TECH-7777
